OpenSim/Examples/Plugins/BodyDragExample/BodyDragForce.cpp

```diff
@@ -115,12 +115,17 @@ void BodyDragForce::computeForce(const SimTK::State& s,
 
     SimTK::Vec3 bodyCoMPosBody, bodyCoMPosGround, bodyCoMVelGround, bodyCoMVelGroundRaisedPower, dragForceGround, dragForceBody, oppVelSign;
     BodySet &bs = _model->updBodySet();                                     // get body set
-    Body &ground = _model->getSimbodyEngine().getGroundBody();              // get ground body
+    const Ground &ground = _model->getGround(); // get ground body
     Body &aBody = bs.get(get_body_name());                                      // get the body to apply the force to
 
-    bodyCoMPosBody = aBody.getMassCenter();                                                 // get CoM position of body in the BODY coordinate system
-    _model->getSimbodyEngine().getPosition(s, aBody, bodyCoMPosBody, bodyCoMPosGround);     // get CoM position of body in the GROUND coordinate system
-    _model->getSimbodyEngine().getVelocity(s, aBody, bodyCoMPosBody, bodyCoMVelGround);     // get CoM velocity of body in the GROUND coordinate system
+    // get CoM position of body in the BODY coordinate system
+    bodyCoMPosBody = aBody.getMassCenter();
+    // get CoM position of body in the GROUND coordinate system
+    bodyCoMPosGround = aBody.getPositionInGround(s);
+    // get CoM velocity of body in the GROUND coordinate system
+    auto spatialVel = aBody.getVelocityInGround(s);
+    bodyCoMVelGround = spatialVel[1] + SimTK::cross(spatialVel[0],
+                                                    bodyCoMPosBody);
```

Comment:
This looks correct to me, but it would be better to replace these two calls with `aBody.findStationVelocityInGround()`, [here](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Simulation/Model/Frame.h#L239).